cryptarchia/test_fork_choice.py

```diff
@@ -8,26 +8,31 @@
 
 
 def make_block(parent_id: Id, slot: Slot, block_id: Id) -> BlockHeader:
-    return BlockHeader(parent=parent_id, id=block_id, slot=slot)
+    assert len(parent_id) == 32
+    assert len(block_id) == 32
+    return BlockHeader(parent=parent_id, content_size=1, slot=slot, content_id=block_id)
 
 
 class TestLeader(TestCase):
     def test_fork_choice_long_sparse_chain(self):
         # The longest chain is not dense after the fork
-        common = [make_block(b"", Slot(i), str(i).encode()) for i in range(1, 50)]
+        common = [
+            make_block(bytes(32), Slot(i), bytes(repeat(i, 32))) for i in range(1, 50)
+        ]
         long_chain = deepcopy(common)
         short_chain = deepcopy(common)
 
         for slot in range(50, 100):
             # make arbitrary ids for the different chain so that the blocks appear to be different
-            long_id = hashlib.sha256(f"{slot}-long".encode()).digest()
-            short_id = hashlib.sha256(f"{slot}-short".encode()).digest()
+            long_id = hashlib.sha256(f"{slot}-long".encode()).digest()[:32]
+            short_id = hashlib.sha256(f"{slot}-short".encode()).digest()[:32]
```

Comment:
Why do we need the `[:32]` on the end here?

Reply:
the digest is by default 64bytes, I could not find an easy way to set that except  constructing a sha256 instance (which this convenience method does under the hood)

Reply:
I'm not seeing 64 bytes here?
```python
>>> len(sha256(b"blah").digest())
32
```
In this looks like a no-op
```python
>>> hash = sha256(b"blah").digest()
>>> hash[:32] == hash
True
```

Reply:
looks like you're right, what was I looking at ?🤔